src/router/users.ts

```diff
@@ -0,0 +1,111 @@
+import { Router } from 'express';
+import Joi from '@hapi/joi';
+import { ValidatedRequestSchema, createValidator, ContainerTypes, ValidatedRequest } from 'express-joi-validation';
+import { User, createUser, updateUser } from '../user/user';
+
+const route = Router();
+const validator = createValidator();
+
+const postUserSchema = Joi.object({
+    login: Joi.string().required(),
+    password: Joi.string()
+        .alphanum()
+        .required(),
+    age: Joi.number()
+        .integer()
+        .min(4)
+        .max(130)
+        .required(),
+});
+
+const putUserSchema = Joi.object({
+    login: Joi.string(),
+    password: Joi.string().alphanum(),
+    age: Joi.number()
+        .integer()
+        .min(4)
+        .max(130),
+});
+
+const autoSuggestUsersSchema = Joi.object({
+    loginSubstring: Joi.string(),
+    limit: Joi.number(),
+});
+
+interface UsersRequestSchema extends ValidatedRequestSchema {
+    [ContainerTypes.Body]: {
+        login: string;
+        password: string;
+        age: number;
+    };
+}
+
+interface AutoSuggestUsersSchema extends ValidatedRequestSchema {
+    [ContainerTypes.Query]: {
+        loginSubstring: string;
+        limit: number;
+    };
+}
+
+const users: Map<string, User> = new Map();
```

Comment:
Will be better to create users store and access objects in another file
Let's keep the rule of single responsibility